python/pyarrow/__init__.py

```diff
@@ -44,6 +44,13 @@ def parse_version(root):
         __version__ = None
 
 
+from pyarrow.compat import import_tensorflow_extension
+
+
+# Workaround for https://issues.apache.org/jira/browse/ARROW-2657
+import_tensorflow_extension()
```

Comment:
Nit: this exposes `import_tensorflow_extension` in the pyarrow.* namespace. I'll fix